Extensions/ShareTo/ShareViewController.swift

```diff
@@ -367,12 +367,14 @@ extension ShareViewController {
         }
 
         gesture.isEnabled = false
-        sendToDevice = SendToDevice()
-        guard let sendToDevice = sendToDevice else { return }
-        sendToDevice.sharedItem = item
-        sendToDevice.delegate = delegate
-        let vc = sendToDevice.initialViewController()
-        navigationController?.pushViewController(vc, animated: true)
+        RustFirefoxAccounts.shared.accountManager.uponQueue(.main) { _ in
+            self.sendToDevice = SendToDevice()
+            guard let sendToDevice = self.sendToDevice else { return }
+            sendToDevice.sharedItem = item
+            sendToDevice.delegate = self.delegate
```

Comment:
Could referencing `self` strongly here cause any retain cycles?

Reply:
Always good practice to use a weak self, costs nothing, protects against retain cycles. Will do it here.

Reply:
actually changed my mind :), these blocks are short-lived and always run to completion so I can't see how they could retain, and I don't want to be inconsistent and do a weak self here when I don't in similar cases.